### PR TITLE
add: `index` param in `selectAsset`

### PR DESCRIPTION
**Pull request by LeGoffMael**
This PR expose grid `index` param in the `selectAsset` delegate function.

### CODE REVIEW

**Comment by AlexV525** [comment]
What is the purpose for the index? This will be a breakimg change.

**Comment by LeGoffMael** [comment]
For my case, the goal is to get the position of the selected asset in the grid view, so i can move the scroll controller to it (same behavior as instagram).

```dart
final row = (index / gridCount).floor();
final size =
    (MediaQuery.of(context).size.width - itemSpacing * (gridCount - 1)) /
        gridCount;
return row * size + (row * itemSpacing);
```

**Comment by AlexV525** [comment]
Please add a changelog, a migration guide, and bump the version to 8.3.0 in both the package and the example (the example requires version code addition too).

**Comment by LeGoffMael** [comment]
done~

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,10 @@ that can be found in the LICENSE file. -->
 
 # Changelog
 
+## 8.3.0
+
+- Add `index` argument to `selectAsset` in `AssetPickerBuilderDelegate`. (#399)
```
Comment (AlexV525) [inline CHANGELOG.md:9]:
```suggestion
### New features

- Add `index` argument to `selectAsset` in the `AssetPickerBuilderDelegate`. (#399)
```

File: guides/migration_guide.md [hunk 2]
```diff
@@ -14,6 +14,10 @@ This document gathered all breaking changes and migrations requirement between m
 - [6.0.0](#600)
 - [5.0.0](#500)
 
+## 8.3.0
+
+Delegates extending `AssetPickerBuilderDelegate` that implements `selectAsset` should add the `index` argument to its signature.
```
Comment (AlexV525) [inline guides/migration_guide.md:19]: You can quote the comment to get the raw text. GitHub seems not support with nested markdown codes.

### Summary

Delegates extending `AssetPickerBuilderDelegate` that implements `selectAsset`
should add the `index` argument to its signature.

### Details

Before:

```dart
void selectAsset(
  BuildContext context,
  Asset asset,
  bool selected,
);
```

After:

```dart
void selectAsset(
  BuildContext context,
  Asset asset,
  int index,
  bool selected,
);
```

**Comment by AlexV525** [comment]
BTW, the picker looks beautiful in the recording. It would be great if you're willing to contribute it as an example in the repo.

**Comment by LeGoffMael** [comment]
I applied your remarks in the migration guide and the changelog.

> BTW, the picker looks beautiful in the recording. It would be great if you're willing to contribute it as an example in the repo.

Thank you~! It actually try to mimic the instagram picker, because i also needed the crop view and a controller i made a repo (still in development): [insta_assets_picker](https://github.com/LeGoffMael/insta_assets_picker)

I did not submit it as a custom delegate because the crop feature seems too out of scope. But if you're interested i could replace the crop view by a simple carousel and submit `InstaAssetPickerBuilder`.

**Comment by AlexV525** [comment]
> But if you're interested i could replace the crop view by a simple carousel and submit `InstaAssetPickerBuilder`.

That would be nice!

**Review by AlexV525 — APPROVED** [review]
LGTM

**Comment by AlexV525** [comment]
@all-contributors Add @LeGoffMael for code.

**Pull request merged**